docs/developer/style-guide.md

```diff
@@ -0,0 +1,119 @@
+
+You can do this proof-reading when creating the PR in GitHub, as there is a nice summary of the changes presented at this time. You can also do it by opening the `Files Changed` tab after the PR has been created.
+
+### Submit a draft if you are unsure {#submit-a-draft}
+
+If you are unsure about you changes, want to get feedback on your code from a co-worker, or simply think your work is not ready yet you should **submit your PR as Draft**. This will signal to the other developers that you are not done.
+
+### Link relevant issues
+
+You should link relevant issues inside your PR body. Use closing keywords to auto-close issues when your PR is merged. See [the GitHub documentation](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) for more info.
+
+## Sharing code in markdown documents, messages etc
+
+When sharing code with co-workers and developers alike please take the time to format it properly. Code that is not formatted is not easy to read and your co-worker will have a hard time de-cyphering what you intended to write.
+
+### For documents and comments where you can specify the language used {#markdown-code-blocks}
+
+GitHub supports markdown nearly everywhere from issue and pr body to review comments. You should always define the language of your code snippets. Define a code block with 3 backticks and add the language after the backticks on the first line, like so:
+
+~~~md
+```ts
+const hello = "world";
+```
+~~~
+
+This will provide syntax highlighting and make the experience of reading code a lot more enjoyable for other people.
+
+:::tip
+
+Read [the Basics of Markdown formatting](https://docs.github.com/en/get-started/writing-on-github/getting-started-with-writing-and-formatting-on-github/basic-writing-and-formatting-syntax) in the GitHub docs if you need a refresher of how to format certain things.
+
+:::
+
+### For messages (Slack etc)
+
+Most instant messaging platforms allow you to use code blocks for sending multiline code in messages. In slack it uses the same 3 backticks syntax as described in [the markdown section](#markdown-code-blocks), but does not support adding the language. So there is no syntax highlighting available.
+
+### Code present in GitHub
+
+Use Github permalinks to reference code snippets in messages. Navigate to the source code that you want to send, click on the line number, click on the 3 dots and select "Copy permalink". This will create a link that you can send to your co-worker to improve communication about bits of code (there will even be a preview generated on the fly, how cool !).
```

Comment:
```suggestion
Use Github permalinks to reference code snippets in messages. Navigate to the source code that you want to send, click on the line number, click on the 3 dots and select "Copy permalink". This will create a link that you can send to your co-worker to improve communication about bits of code (there will even be a preview generated on the fly, how cool!).
```